twisted/conch/test/test_channel.py

```diff
@@ -151,8 +151,22 @@ def test_str(self):
         Test that str(SSHChannel) works gives the channel name and local and
         remote windows at a glance..
         """
-        self.assertEqual(str(self.channel), '<SSHChannel channel (lw 131072 '
-                'rw 0)>')
+        if _PY3:
+            self.assertEqual(
+                str(self.channel), "<SSHChannel b'channel' (lw 131072 rw 0)>")
+
+        else:
+            self.assertEqual(
+                str(self.channel), '<SSHChannel channel (lw 131072 rw 0)>')
+
+    def test_bytes(self):
+        """
+        Test that bytes(SSHChannel) works, gives the channel name and
+        local and remote windows at a glance..
+
+        """
+        self.assertEqual(
+            bytes(self.channel), b'<SSHChannel channel (lw 131072 rw 0)>')
```

Comment:
bytes(self.channel) behaves differently on Python 2 and Python 3.
On Python 3, bytes(self.channel) will call the self.channel.__bytes__() method if it exists.
On Python 2, it does not.

```
class A:

    def __bytes__(self):
        print("I called __bytes__()")
        return b"Hi"


a = A()

b = bytes(a)
```

Until Twisted drops support for Python 2, I advise to explictly call __bytes__() to have consistent behavior.